doc/user/layouts/partials/create-source/connector/s3/details.html

```diff
@@ -1,22 +1,76 @@
 ### S3 source details
 
-The S3 source is designed to ingest a large volume of static data from S3.
+The S3 source is designed to ingest a large volume of static data from [AWS's Simple Storage
+Service][aws-s3]. S3 is Amazon Web Services' cloud object store. It is important to note that an
+object store behaves differently than a file system does, and that those differences affect how
+Materialize interacts with objects in it.
+
+Some key differences from file systems include:
+
+* Latency on individual S3 operations is much higher than even cloud-network filesystems.
+* It is not possible to seek to a point in an object, the entire object must be downloaded from
+  the beginning in order to read to a point.
+* All object operations are atomic, and it is not possible to modify just part of an S3 object,
+  you can only entirely replace or delete objects.
```

Comment:
I would break this into "All objects are atomic. It is not possible to modify just part of an S3 object; you can only entirely replace or delete objects." (Note comma needs to be a semicolon even if you keep this as a single sentence.)